### PR TITLE
Feat/GWW-126: use datepickers on the comparison map

### PR DESCRIPTION
**Pull request by lsarni**
Ticket: [GWW-126](https://issuetracker.deltares.nl/browse/GWW-126)

**Main features:**
- Replace the date with arrows for date pickers on the comparison map
- Only allow selecting dates on the time series
- Override accent color for all date pickers
- The date picker adjust it's positioning to be always fully visible

image

### CODE REVIEW

File: src/components/ComparisonMap/ComparisonMap.vue [hunk 1]
```diff
@@ -100,45 +114,32 @@
 
     data () {
       return {
-        dateIndex: -1,
-        oldDateIndex: -1,
+        isoDate: (new Date()).toISOString().substring(0, 10),
```
Comment (SiggyF) [inline src/components/ComparisonMap/ComparisonMap.vue:117]: Can you explain this?

File: src/components/ComparisonMap/ComparisonMap.vue [hunk 2]
```diff
@@ -159,14 +160,16 @@
         // On mounted, `date` is the last date in the time series
         // `oldDate` is the closest date in the time series one year before `date`
         const date = this.timeSeriesDates[this.timeSeriesDates.length - 1]
+        this.isoDate = date.toISOString().substring(0, 10)
```
Comment (SiggyF) [inline src/components/ComparisonMap/ComparisonMap.vue:163]: You can also use a custom format.

**Review by SiggyF — APPROVED** [review]
Looks good, some minor questions.

File: src/components/ComparisonMap/ComparisonMap.vue (hunk 2 above)
Comment (lsarni) [reply src/components/ComparisonMap/ComparisonMap.vue:163]: For a custom format I would [need a library](https://stackoverflow.com/questions/1056728/where-can-i-find-documentation-on-formatting-a-date-in-javascript) like Moment.js

File: src/components/ComparisonMap/ComparisonMap.vue (hunk 1 above)
Comment (lsarni) [reply src/components/ComparisonMap/ComparisonMap.vue:117]: The Vuetify date picker [accepts ISO 8601 date strings (YYYY-MM-DD)](https://v2.vuetifyjs.com/en/components/date-pickers/#caveats). It was quite unexpected for me. I will see what is the best way to add a comment for that.

**Pull request merged**